srtcore/core.cpp

```diff
@@ -7790,34 +7773,42 @@ int CUDT::processData(CUnit* unit)
    CGuard::leaveCS(m_StatsLock);
 
    {
-      /*
-      * Start of offset protected section
-      * Prevent TsbPd thread from modifying Ack position while adding data
-      * offset from RcvLastAck in RcvBuffer must remain valid between seqoff() and addData()
-      */
+      // Start of offset protected section
+      // Prevent TsbPd thread from modifying Ack position while adding data
+      // offset from RcvLastAck in RcvBuffer must remain valid between seqoff() and addData()
       CGuard recvbuf_acklock(m_AckLock);
 
-      int32_t offset = CSeqNo::seqoff(m_iRcvLastSkipAck, packet.m_iSeqNo);
+      const int offset = CSeqNo::seqoff(m_iRcvLastSkipAck, packet.m_iSeqNo);
 
       bool excessive = false;
-      string exc_type = "EXPECTED";
-      if ((offset < 0))
+#ifdef ENABLE_HEAVY_LOGGING
```

Comment:
Kinda overspoken. In one of the works I created somewhere a type named `DebugString`, which turned into a nothing and ignoring all operations if compiling without heavy logging. In this particular case maybe it would be easier something like this:
```
#if ENABLE_HEAVY_LOGGING
#define IF_HEAVY_LOGGING(instr) instr
#else 
#define IF_HEAVY_LOGGING(instr) (void)0
#endif 
```